### PR TITLE
Ensure that layout commit messages are received by quorum

### PR DESCRIPTION
**Pull request by StrikingThirteen**
## Overview

Description:
This ensures that layout commit are received by quorum.

Why should this be merged: 
Make sure that quorum receives the commit message to resolve an issue where servers missed the commit messages.
 
Related issue(s) (if applicable): #<number>

## Checklist (Definition of Done):

- [x] There are no TODOs left in the code
- [x] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [x] Change is covered by automated tests
- [x] Public API has Javadoc

### CODE REVIEW

File: runtime/src/main/java/org/corfudb/runtime/view/LayoutView.java [hunk 1]
```diff
@@ -301,14 +303,21 @@ public void committed(long epoch, Layout layout, boolean force)
...
 
+        // Make sure a quorum gets to commit.
         int timeouts = 0;
-        int responses = 0;
-        while (responses < commitList.length) {
-            // wait for someone to complete.
+        long wrongEpochRejected = 0L;
+        while (true) {
+            if (commitList.length < getQuorumNumber()) {
+                log.debug("commit: Quorum unreachable, remaining={}, required={}", commitList,
```
Comment (Maithem) [inline runtime/src/main/java/org/corfudb/runtime/view/LayoutView.java]: commitList is based on a layout passed to the method, which can be different from the layout used by getQuorumNumber

File: runtime/src/main/java/org/corfudb/runtime/view/LayoutView.java [hunk 2]
```diff
@@ -301,14 +303,21 @@ public void committed(long epoch, Layout layout, boolean force)
                 })
                 .toArray(CompletableFuture[]::new);
 
+        // Make sure a quorum gets to commit.
         int timeouts = 0;
-        int responses = 0;
-        while (responses < commitList.length) {
-            // wait for someone to complete.
+        long wrongEpochRejected = 0L;
+        while (true) {
```
Comment (Maithem) [inline runtime/src/main/java/org/corfudb/runtime/view/LayoutView.java]: Why spin here, shouldn't this be a best effort? the fault detector should periodically patch the management layout, commited layout and epochs

Comment (medhavidhawan) [reply runtime/src/main/java/org/corfudb/runtime/view/LayoutView.java]: As of now the fault detector does not do that and hence this patch with FIXMEs

File: runtime/src/main/java/org/corfudb/runtime/view/LayoutView.java (hunk 1 above)
Comment (medhavidhawan) [reply runtime/src/main/java/org/corfudb/runtime/view/LayoutView.java]: Yep this is wrong and should be fixed.

File: runtime/src/main/java/org/corfudb/util/retry/IntervalRetry.java [hunk 3]
```diff
@@ -22,7 +21,7 @@
      **/
     @Getter
     @Setter
-    Duration retryInterval = Duration.ofMillis(1000);
+    long retryInterval = 1000;
```
Comment (xnull) [inline runtime/src/main/java/org/corfudb/util/retry/IntervalRetry.java]: Why it's changed to long?

File: runtime/src/main/java/org/corfudb/runtime/view/LayoutView.java [hunk 4]
```diff
@@ -317,12 +326,31 @@ public void committed(long epoch, Layout layout, boolean force)
...
+
+            // Exclude from the commit list those that completed exceptionally.
             commitList = Arrays.stream(commitList)
                     .filter(x -> !x.isCompletedExceptionally())
                     .toArray(CompletableFuture[]::new);
 
-            log.debug("committed: Successful responses={}, timeouts={}", responses, timeouts);
+            // Count successes.
+            long count = stream(commitList)
+                    .map(x -> {
```
Comment (xnull) [inline runtime/src/main/java/org/corfudb/runtime/view/LayoutView.java]: can you rename `x` to something meaningful

**Comment by Maithem** [comment]
superseded by #1660

**Pull request closed without merging**